proposed/http-middleware/middleware-meta.md

```diff
@@ -1,49 +1,56 @@
-PSR-15 Meta Document
-====================
+HTTP Server Middleware
+======================
 
 1. Summary
 ----------
 
 The purpose of this PSR is to provide an interface that defines the formal
-method signature for HTTP Middleware that is compatible with HTTP Messages,
-as defined in PSR-7.
+method signature for HTTP Server Middleware that is compatible with HTTP
+Messages, as defined in [PSR-7][psr7].
+
+[psr7]: http://www.php-fig.org/psr/psr-7/
 
 2. Why Bother?
 --------------
 
-The general concept of reusable middleware was popularized by [StackPHP][stackphp].
-Since the release of the HTTP Messages standard, a number of frameworks have
-adopted middleware that uses HTTP Message interfaces.
+The HTTP Messages specification does not contain any reference to HTTP Middleware.
+
+The design pattern used by middleware has existed for many years as the
+[pipeline pattern][pipeline], or more specifically, "linear pipeline processing".
```

Comment:
@shadowhand Do not try to distract from real issue:

**Reviewers can neither know who is responsible for that nor does they know how it was possible that this sentence got here.**

---

While this is totally off-topic to that issue, I don't know what you mean too.
1. The  [Wikipedia article](https://en.wikipedia.org/wiki/Pipeline_(computing)) does not contain the word middleware at all.
2. The meaning of the sentence is not: _middleware is an **example of** pipeline processing_, is it?
3. _linear pipeline processing_ is not a [(software) design pattern](https://en.wikipedia.org/wiki/Software_design_pattern), and the [Wikipedia article](https://en.wikipedia.org/wiki/Pipeline_(computing)) does not contain the word _pattern_ either (there is only one unrelated exception)
4. The middleware pattern we are talking about is rather new compared to pipeline processing.
5. _The design pattern used by middleware has existed for many years_:<br>While this is might be true for the Functional Programming Pattern, this is not true for the interfaces you propose. Certainly nobody, have had ever the idea to translate a _Functional Programming Pattern_ into a pure _Object Oriented Pattern_ in such a way. You may ask why. Simply because:
   1. If you use well established Object Oriented Analysis and Design techniques, then it is highly unlikely that you will get those interfaces.
   2. The [Chain of Responsibility Pattern](https://sourcemaking.com/design_patterns/chain_of_responsibility) is well known for many many years, which uses only 1&nbsp;interface (not 2!) and fit much much better into the OOP world (which supports my argument at _i._).